Update server documentation: auto import/exports caveats

PrimeTimeTran opened the pull request:
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add additional context on what is auto imported/exported from `./server` so that future developers know the caveats of auto import/export

<img width="1035" alt="Screenshot 2023-11-15 at 10 42 58 PM" src="https://github.com/nuxt/nuxt/assets/16382165/6e541517-c5ab-415f-9dc3-0349a8b6314c">


I see builds failing in CI/CD when I change casing from `./server/models/models.model.ts` to `./server/Models/models.model.ts`

Please let me know if this shouldn't be the case.

### 📝 Checklist

- [x] I have updated the documentation accordingly.

PrimeTimeTran committed:
Update 1.server.md

Add additional documentation about auto import/exports from `./server/*`

PrimeTimeTran committed:
Merge pull request #1 from PrimeTimeTran/patch-1

Update 1.server.md

PrimeTimeTran committed:
Fix example

huang-julien commented on docs/2.guide/2.directory-structure/1.server.md:
```diff
@@ -494,3 +494,22 @@ export default defineNuxtConfig({
+
+
+
+### Cavets on Auto import/export
```
```suggestion
### Caveats on Auto import/export
```

huang-julien commented on docs/2.guide/2.directory-structure/1.server.md:
````diff
@@ -494,3 +494,22 @@ export default defineNuxtConfig({
+
+`Wizard` is exported from this file automatically.
+
+```ts [./server/apis/wizards.get.ts]
````
This is the same name as the one bellow, did you meant utils ?

PrimeTimeTran replied:
My bad, I meant to specifically point out that Models & models are different/case sensitive from my observation so far.

huang-julien reviewed (changes requested):
I don't think exported variables from api files are auto-imported in other files

danielroe reviewed (commented):
I'm not sure this is needed. We already have this explanation: https://nuxt.com/docs/guide/directory-structure/server#server-utilities.

Maybe we could add another sentence to make it clearer?

PrimeTimeTran committed:
Update docs/2.guide/2.directory-structure/1.server.md

Co-authored-by: Julien Huang <julien.h.dev@gmail.com>

PrimeTimeTran committed:
Update example

PrimeTimeTran committed:
Update example

PrimeTimeTran commented:
> I'm not sure this is needed. We already have this explanation: https://nuxt.com/docs/guide/directory-structure/server#server-utilities.
> 
> Maybe we could add another sentence to make it clearer?

I updated the example & also added another interesting thing I discovered.

https://s5.gifyu.com/images/SRHNy.gif

Type suggestions work when a file exports an additional const. Not sure if this is the expected behavior but it caught me off guard and threw me off.

PrimeTimeTran commented:
@danielroe please let me know if I've been unclear. I'll be more than happy to update to land this.

My thinking: 
- Auto import/export is awesome but isn't defined from what I've found so far.  
- Type definitions in one repo for both front and backend solves a big problem but not getting autocomplete/suggestions due to a missing const export seems strange to me.

Please correct me if I'm wrong. I'm hoping to save future users time wondering why they're not getting them(type suggestions) and their local auto imports work but fail in CI/CD.

I'm on `"nuxt": "^3.8.0",` in case that makes a difference.

danielroe commented:
I think type imports failing in CI is likely due to a failure to run `nuxi prepare` which generates the `tsconfig.json`.

We should specify that files in `server/utils/` are scanned for auto-imports, [here](https://nuxt.com/docs/guide/directory-structure/server#server-utilities).

danielroe commented:
Marking as draft just to help me triage PRs better... Feel free to unmark whenever you're ready for me to have a look.

danielroe commented:
I think as written, these aren't suggestions we'd want to make.

- casing of models/Models shouldn't make a difference to _nuxt_ as we don't do anything special with that directory. But if you are importing from it, you will encounter issues when developing on a case-insensitive OS when you then build in CI. But this isn't related to auto imports.
- As for auto-import of types, this isn't yet supported. Likely this is working for you by accident and we wouldn't want to document the current solution.

Thank you for taking the time to create the PR however, and please do feel free to create another one any time ❤️
